### PR TITLE
Revert "Fixed Camera shifting Bug"

### PR DESCRIPTION
**Pull request by bvssvni**
Reverts PistonDevelopers/hematite#250

See https://github.com/PistonDevelopers/hematite/pull/250#discussion_r46778735.

### CODE REVIEW

**Comment by RichardlL** [comment]
@bvssvni 

Except it causes the bug because the camera height shouldn't be determined by camera/ movement (which is why the bug happens, but should happen here 211/212
````
let mut camera = first_person.camera(0.0);
                camera.position[1] += 1.62;
but instead add camera.position[1] +=1.62 + players position
````

**Comment by RichardlL** [comment]
O.k. It looks like vec3_scales in piston are used for calculating angle for viewing sometime, respective of camera position . (Like if you had a moon and you always wanted the moons face looking at the earth) or vise versa.

In minecraft we don't use this, as camera angle isn't dependent on movement and movement isn't dependent on on camera angle

e.g. 
moving up-left and you still are looking forwards

We don't need this in minecraft, as camera is not dependent on movement

The camera's y should be set 1.62 above the players y position. (assuming that's its feet)

What makes it even more confusing is it seems x and y in piston for camera position, so @eddyb is right, the whole thing is unneeded.

**So what seems what was happening

Camera X (relative, meaning the camera is moving forward) .01 * the yaw change of the camera
It seems to be due to only happen near the axis's due to rounding,
Changing the value from .1 to 5 confirms this, This effectively is making the player xy position move when the camera yaw and pitch moves, yet it does nothing, normally due to rounding, but occurs at extremes near the axis
This is not intended behavior, as we are not orbiting anything
So put in layman's terms...

The camera is moving forwards when looking down, or backwards when looking up.

This is used in places with orbiting, such as 3rd person.

For first person, this shouldn't happen at all, since the players eyes are centered, and where not staring at something .1 meters in front of us.

If you change the value to 5 (from 0.1), it puts you in " third person", you're orbiting your players postion in 3rd person, and that eliminates the rounding problem.

This is how third person in minecraft works, as the camera is just orbiting 1.62 above your position instead of being 1.62 above your location, and your body is rendered too.

Changing this to 0.0 fixes the bug due to nullifying the effects entirely.

Deleting this doesn't remove the feature, it only removes the bug.

Camera XYY should be players X, Y+1.62 ,Z.

This doesn't do that, it makes the camera orbit something .1 meters away from the player

**Comment by RichardlL** [comment]
Additionally, we wont be able to set the cameras Y, until time/acceleration/non freecam  is implemented, due to jumping/falling. Right now the camera is just a position, and moved when controls are moved, where in minecraft it takes the players x and y, it jumps at a certain rate, and then falls until the players y is 1.62 above the highest block below  the player.

also, Is there a reason the players x,y,z is not just the eyes in the first place? That's the way minecraft does it. Open up minecraft and the Y will always be Y.6 something.

**Comment by eddyb** [comment]
The eyes in Minecraft are *definitely not* centered, they are offset.

**Comment by RichardlL** [comment]
Oh ok, now its starting to make sense

I assumed the off set was unintentional, and that you were trying to achieve something else

But now the bug is that it isn't that the camera is shifting at the axis extremes, but that due to rounding, it **only** shifts at the extremes.

So effectively, it doesn't offset, but when you look straight up or down it does, Which is what causes the jerk.

**Comment by eddyb** [comment]
What happens is that the x and z components of `camera.forward` are *exactly* 0 when looking straight up or down, which is why using `camera.forward` is wrong here, the values should be the sin/cos of the actual yaw
